docs/tool_interface.rst

```diff
@@ -11,50 +11,60 @@ In order to describe the interface, a few definitions are required at first.
+    Description of a given state of the user interface: what should the user be able to interact with in that *view*, which other *view* can the user visit from the current *view*. The collection of *views* form together the user interface. Note: the view does not describe how the interface looks like to the user, this is described in the *view-rendering*.
 
 .. _view-component-def:
 
-view-component
+View-component
 --------------
-    A *view-component* is a part of a *view* which can be described independently from the view and could be reused in different *views* (a menu bar for example). A *view-component* is described in terms of its attributes and its actions. For example a menu bar has a nested list of items (each item can itself be a list of item: submenus). Each item, if not itself a list, consists of a label (what the user will see) and an action (what will be done upon clicking/selecting the item.)
+    A *view-component* is a part of a *view* which can be described independently from the view and could be reused in different *views* (a menu bar for example). A *view-component* is described in terms of its attributes and its actions. For example a menu bar has a nested list of items (each item can itself be a list of item: submenus). Each item, if not itself a list, consists of a label (what the user will see) and an action (what will be done upon clicking/selecting the item).
 
 .. _attribute-def:
 
 Attribute
 ---------
-    The attributes contain the essential information needed to characterize and render a :ref:`view-def` or a :ref:`view-component-def`. Some attributes might be rendered themselves in the :ref:`view-def` or :ref:`view-component-def` but it is not required. Some attributes can be bound to :ref:`requirement-def` of :ref:`action-def`.
+    An attribute contains the essential information needed to characterize and render a :ref:`view-def` or a :ref:`view-component-def`. An attribute might be visible in the :ref:`view-def` or :ref:`view-component-def` but it can also just be information. An attribute can be linked to a certain number of :ref:`action-def` and to certain :ref:`requirement-def`.
+
+.. _action-def:
+
+Action
+------
+    An action describes what the user can do in a certain :ref:`view-def` or :ref:`view-component-def`. Each action has an index which is used to link it to an :ref:`attribute-def`.
+
+.. _requirement-def:
+
+_Requirement
+------------
+    A requirement describes necessary features of a :ref:`view-def`, a :ref:`view-component-def` or a :ref:`action-def`. A requirement can specify required actions by the user as well as describing what happens if a requirement is not met. Each action has an index which is used to link it to an :ref:`attribute-def`.
```

Comment:
```suggestion
    A requirement describes specific conditions of a :ref:`view-def`, a :ref:`view-component-def`, an :ref:`action-def` or an :ref:`attribute-def` which need to be met. Each requirement has an index which is used to link it to an :ref:`attribute-def`.
```

Reply:
I would provide an example of a simple requirement and a simple action to help the readers

Reply:
Examples

Action: the user can click on button A to trigger action B

Requirement: the button B cannot be clicked unless text input C is not filled by user